### ISSUE TITLE
`hash_pubkeys` Serializes Compressed Public Keys Instead of Uncompressed

### ISSUE THREAD

**Issue by codingkindi** [jurraca/cashulib#3]
The `hash_pubkeys` function in the Elixir implementation serializes **compressed** public keys, while the reference Python implementation serializes **uncompressed** keys. Public keys should be serialized in **uncompressed** format for consistency with the reference implementation.

**Comment by jurraca** [comment]
As noted in #4 , [NUT-01](https://github.com/cashubtc/nuts/blob/main/01.md#keyset-generation) specifies compressed public keys for mint pubkeys. Blind points are also compressed. Could you share the code in the nutshell impl? i suspect it's to support an old cashu version.

**Comment by codingkindi** [comment]
Certainly! If you review the current implementation of the `hash_e` function, located at the following path:

[https://github.com/cashubtc/nutshell/blob/main/cashu/core/crypto/b_dhke.py](https://github.com/cashubtc/nutshell/blob/main/cashu/core/crypto/b_dhke.py)

you’ll see that the `serialize` method is explicitly called with `compressed=False`, ensuring an uncompressed serialization of each `PublicKey` object:

```python
def hash_e(*publickeys: PublicKey) -> bytes:
    e_ = ""
    for p in publickeys:
        _p = p.serialize(compressed=False).hex()
        e_ += str(_p)
    e = hashlib.sha256(e_.encode("utf-8")).digest()
    return e
```

This implementation concatenates the uncompressed, hex-encoded public keys and then hashes the result with SHA-256.

**Comment by jurraca** [comment]
hmm interesting. This is for the DLEQ [calculation](https://github.com/cashubtc/nuts/blob/main/12.md#hashx-arraypublickey---bytes) though, not the general representation of public keys. Curious why that is.

**Comment by codingkindi** [comment]
I can see that there is changes requested in the PR but I'm not sure how to do it. Can you guide me ?

**Comment by jurraca** [comment]
left a comment on the PR

**Comment by codingkindi** [comment]
Yea, I will reply 👍
